@xen-orchestra/lite/src/views/vm/VmRootView.vue

```diff
@@ -1,24 +1,24 @@
 <template>
   <ObjectNotFoundWrapper object-type="vm">
     <VmHeader />
+    <VmTabBar :uuid="vm!.uuid" />
     <RouterView />
   </ObjectNotFoundWrapper>
 </template>
 
 <script lang="ts" setup>
 import ObjectNotFoundWrapper from "@/components/ObjectNotFoundWrapper.vue";
 import VmHeader from "@/components/vm/VmHeader.vue";
-import { watchEffect } from "vue";
-import { useRoute } from "vue-router";
+import VmTabBar from "@/components/vm/VmTabBar.vue";
 import { useUiStore } from "@/stores/ui.store";
 import { useVmStore } from "@/stores/vm.store";
+import { whenever } from "@vueuse/core";
+import { computed } from "vue";
+import { useRoute } from "vue-router";
 
 const route = useRoute();
 const vmStore = useVmStore();
 const uiStore = useUiStore();
-
-watchEffect(() => {
-  const vm = vmStore.getRecordByUuid(route.params.uuid as string);
-  uiStore.currentHostOpaqueRef = vm?.resident_on;
-});
+const vm = computed(() => vmStore.getRecordByUuid(route.params.uuid as string));
+whenever(vm, (vm) => (uiStore.currentHostOpaqueRef = vm.resident_on));
```

Comment:
```js
whenever(vm !== undefined, ...)
```
Should we make the condition explicit?

Reply:
`vm` is a `computed`, so `vm !== undefined` would always returns `true`.

In addition, `whenever` watches reactive data and run the callback _whenever_ it is truthy, passing the resulting value as first argument.

So `whenever(() => vm.value !== undefined, (vm) => {})` would pass the value `true` as first argument instead of a `XenApiVm`